clayui.com/src/components/Hooks/useStateWithLocalStorage.js

```diff
@@ -5,45 +5,17 @@
 
 import React from 'react';
 
-const inBrowser = typeof window !== 'undefined';
-
-const localStorage = {
-	getItem(key) {
-		try {
-			return inBrowser ? window.localStorage.getItem(key) : null;
-		} catch {
-			return null;
-		}
-	},
-
-	setItem(key, value) {
-		if (inBrowser) {
-			try {
-				window.localStorage.setItem(key, value);
-			} catch {
-				return;
-			}
-		}
-	},
-};
+import localStorage from '../../utils/localStorage';
 
 function useStateWithLocalStorage(defaultValue, key) {
 	const [value, setValue] = React.useState(() => {
-		try {
-			const stickyValue = localStorage.getItem(key);
+		const stickyValue = localStorage.getItem(key);
 
-			return stickyValue === null
-				? defaultValue
-				: JSON.parse(stickyValue);
-		} catch {
-			return defaultValue;
-		}
+		return stickyValue === null ? defaultValue : JSON.parse(stickyValue);
```

Comment:
Minor nit: `JSON.parse` could `throw` if garbage ever got into localStorage (and you have to treat localStorage as "untrusted input" because a user can stick whatever they want in there), so you probably still want a `try`/`catch` here.

`</paranoia>`